File: benchmarks/node/15_gc_pressure_tree.mjs
function build_tree(depth) {
    if (depth === 0) return { value: 0, left: null, right: null };
    const left = build_tree(depth - 1);
    const right = build_tree(depth - 1);
    return { value: left.value + right.value + 1, left, right };
}

function checksum(node) {
    if (node === null) return 0;
    return node.value + checksum(node.left) + checksum(node.right);
}

function gc_pressure_tree(depth, iterations) {
    let total = 0;
    for (let i = 0; i < iterations; i++) {
        const tree = build_tree(depth);
        total += tree.value;
    }
    return total;
}
console.log(gc_pressure_tree(20, 5));

File: benchmarks/python/15_gc_pressure_tree.py
import sys
sys.setrecursionlimit(1100000)

class TreeNode:
    __slots__ = ('value', 'left', 'right')
    def __init__(self, value, left, right):
        self.value = value
        self.left = left
        self.right = right

def build_tree(depth):
    if depth == 0:
        return TreeNode(0, None, None)
    left = build_tree(depth - 1)
    right = build_tree(depth - 1)
    return TreeNode(left.value + right.value + 1, left, right)

def gc_pressure_tree(depth, iterations):
    total = 0
    for i in range(iterations):
        tree = build_tree(depth)
        total += tree.value
    return total
print(gc_pressure_tree(20, 5))
